### ISSUE TITLE
Looking through the clox single-pass compiler peephole: Section 17.1

### ISSUE THREAD

**Issue by cm1776** [munificent/craftinginterpreters#558]
As explained in Section 17.1, "the compiler only has a peephole view into the user’s program while generating code".

In section 17.2, we see the Parser structure which the compiler uses to keep track of the tokens being parsed. This structure has only two fields for storing tokens: previous and current.

As there is no peek function, the compiler has access to only two tokens at any given time. Thus when parsing the following declaration: " var x; ", the compiler initially has access to "var" (which it stores in parser.previous), and "x" (which it stores in parser.current). 

In other words, the peep-hole view of the above declaration is "var x".

If I am right, then an update is required to the image below (showing a key-hole).

![image](https://user-images.githubusercontent.com/44275568/69907778-3029cc80-13dc-11ea-82d3-863d2dead018.png)

The semi-colon in said image would need to be gently air-brushed out. What a pity!

**Comment by cm1776** [comment]
For some strange reason, I could not paste the image of the key-hole. The link to that image is this: https://craftinginterpreters.com/image/compiling-expressions/keyhole.png

**Comment by munificent** [comment]
Heh, this is true, but perhaps we don't need to take this illustration *quite* that literally. :)
